File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/WindowsService/PscuLogFileTransformServiceSettings.cs
using Psi.Business.ServiceContracts.RequestResponse.WindowsServices;

namespace Psi.Data.Models.ClientConfigurationModels.WindowsService
{
    public class PscuLogFileTransformServiceSettings : SettingsBaseHelper
        {
            
            public PscuLogFileTransformServiceSettings(ISettingsBase settingsBase) : base(settingsBase)
            {
            }

            /// <summary>
            /// Retrieves all the configs used by the PscuLogFileTransform Service
            /// </summary>
            public PscuLogFileTransformServiceResponse Configuration()
            {
                var config = new PscuLogFileTransformServiceResponse();

                config.InputFileFields = InputFileFields;
                config.Filters = Filters;
                config.OutputFileFields = OutputFileFields;
                config.PathConfiguration = PathConfiguration;

                return config;
            }

            /// <summary>
            /// The Json configuration for the PscuLogFileTransform Service of PSIWindowsService. This is a JSON object and should be formatted like so:
            /// <remarks>
            /// [{
            ///     "Name": "Transaction Post Date",
            ///     "ValuesCausingInclusion": "",
            ///     "ValuesCausingExclusion": "",
            ///     "RequiresValue": true,
            ///     "ErrorMessage": "Transaction Post Date is required"
            /// }]
            /// </remarks>
            /// </summary>
            [SettingKey("PsiServices.PscuLogFileTransformService.Filters")]
            public string Filters
            {
                get
                { return GetValue(); }
                set { SetValue(value); }
            }

            /// <summary>
            /// The Json configuration for the PscuLogFileTransform Service of PSIWindowsService. This is a JSON object and should be formatted like so:
            /// <remarks>
            /// [{
            ///     "Name": "Savings Account Number",
            ///     "DataType": "System.String",
            ///     "EmptyAllowed": false
            /// }]
            /// </remarks>
            /// </summary>
            [SettingKey("PsiServices.PscuLogFileTransformService.InputFileFields")]
            public string InputFileFields
            {
                get
                { return GetValue(); }
                set { SetValue(value); }
            }

            /// <summary>
            /// The Json configuration for the PscuLogFileTransform Service of PSIWindowsService. This is a JSON object and should be formatted like so:
            /// <remarks>
            /// [{
            ///     "Position": 0,
            ///     "WhitespaceLength": 0,
            ///     "Name": "Savings Account Number",
            ///     "CustomFormatter": null,
            ///     "TruncateToLength": 9,
            ///     "TruncateFromPosition": "",
            ///     "MinimumOutputLength": 9,
            ///     "MinimumOutputPadFromPosition": "",
            ///     "OutputFormatString": "",
            ///     "StaticTextValue": ""
            /// }]
            /// </remarks>
            /// </summary>
            [SettingKey("PsiServices.PscuLogFileTransformService.OutputFileFields")]
            public string OutputFileFields
            {
                get
                { return GetValue(); }
                set { SetValue(value); }
            }

            /// <summary>
            /// The Json configuration for the PscuLogFileTransform Service of PSIWindowsService. This is a JSON object and should be formatted like so:
            /// <remarks>
            /// {
            ///     "InputPath": "input",
            ///     "InputFilenamePattern": "\\.csv$",
            ///     "OutputPath": "output",
            ///     "ErrorPath": "error",
            ///     "ProcessedPath": "processed",
            ///     "CompletedPath": "completed",
            ///     "OutputFilePrefix": "tranlist.cc.",
            ///     "InputFileExclusiveAccessTimeout": "00:05:00"
            /// }
            /// </remarks>
            /// </summary>
            [SettingKey("PsiServices.PscuLogFileTransformService.PathConfiguration")]
            public string PathConfiguration
            {
                get
                { return GetValue(); }
                set { SetValue(value); }
            }
        }
}

File: workspace/legacy-analyzer/output/infrastructure/models/WindowsService/PscuLogFileTransformServiceSettings.ts
import { SettingKey } from '../../decorators/settingKey';
import { 
    PscuFiltersSetting,
    PscuInputFieldsSetting,
    PscuOutputFieldsSetting,
    PscuPathsSetting
} from './PscuSettings';

// Generated interfaces for JSON schema types
export interface FilterConfig {
    Name: string;
    ValuesCausingInclusion: string;
    ValuesCausingExclusion: string;
    RequiresValue: boolean;
    ErrorMessage: string;
}

export interface InputFieldConfig {
    Name: string;
    DataType: string;
    EmptyAllowed: boolean;
}

export interface OutputFieldConfig {
    Position: number;
    WhitespaceLength: number;
    Name: string;
    CustomFormatter: string | null;
    TruncateToLength: number;
    TruncateFromPosition: string;
    MinimumOutputLength: number;
    MinimumOutputPadFromPosition: string;
    OutputFormatString: string;
    StaticTextValue: string;
}

export interface PathConfig {
    InputPath: string;
    InputFilenamePattern: string;
    OutputPath: string;
    ErrorPath: string;
    ProcessedPath: string;
    CompletedPath: string;
    OutputFilePrefix: string;
    InputFileExclusiveAccessTimeout: string;
}

/**
 * Settings for the PSCU Log File Transform Service
 */
export class PscuLogFileTransformServiceSettings {
    private readonly _filters = new PscuFiltersSetting();
    private readonly _inputFields = new PscuInputFieldsSetting();
    private readonly _outputFields = new PscuOutputFieldsSetting();
    private readonly _paths = new PscuPathsSetting();

    /**
     * Filter configurations for the transform service.
     */
    get filters(): FilterConfig[] {
        return this._filters.value;
    }

    set filters(value: FilterConfig[]) {
        this._filters.value = value;
    }

    /**
     * Input field configurations for the transform service.
     */
    get inputFileFields(): InputFieldConfig[] {
        return this._inputFields.value;
    }

    set inputFileFields(value: InputFieldConfig[]) {
        this._inputFields.value = value;
    }

    /**
     * Output field configurations for the transform service.
     */
    get outputFileFields(): OutputFieldConfig[] {
        return this._outputFields.value;
    }

    set outputFileFields(value: OutputFieldConfig[]) {
        this._outputFields.value = value;
    }

    /**
     * Path configurations for the transform service.
     */
    get paths(): PathConfig {
        return this._paths.value;
    }

    set paths(value: PathConfig) {
        this._paths.value = value;
    }

    /**
     * Get all settings as an array of Setting objects
     */
    toSettings() {
        return [
            this._filters.toSetting(),
            this._inputFields.toSetting(),
            this._outputFields.toSetting(),
            this._paths.toSetting()
        ];
    }

    /**
     * Update settings from an array of Setting objects
     */
    fromSettings(settings: Setting[]) {
        for (const setting of settings) {
            switch (setting.key) {
                case this._filters.toSetting().key:
                    this._filters.fromSetting(setting);
                    break;
                case this._inputFields.toSetting().key:
                    this._inputFields.fromSetting(setting);
                    break;
                case this._outputFields.toSetting().key:
                    this._outputFields.fromSetting(setting);
                    break;
                case this._paths.toSetting().key:
                    this._paths.fromSetting(setting);
                    break;
            }
        }
    }
}
